core/new-gui/src/app/dashboard/service/workflow-executions/workflow-executions.service.ts

```diff
@@ -1,16 +1,32 @@
 import { Injectable } from "@angular/core";
-import { Observable } from "rxjs";
+import { BehaviorSubject, Observable } from "rxjs";
 import { AppSettings } from "../../../common/app-setting";
 import { HttpClient } from "@angular/common/http";
 import { WorkflowExecutionsEntry } from "../../type/workflow-executions-entry";
+import { Workflow } from "../../../common/type/workflow";
+import { filter, map } from "rxjs/operators";
+import { WorkflowUtilService } from "../../../workspace/service/workflow-graph/util/workflow-util.service";
+import { WorkflowActionService } from "src/app/workspace/service/workflow-graph/model/workflow-action.service";
+import { UndoRedoService } from "src/app/workspace/service/undo-redo/undo-redo.service";
+import { WorkflowPersistService } from "src/app/common/service/workflow-persist/workflow-persist.service";
+import { Router } from "@angular/router";
 
 export const WORKFLOW_EXECUTIONS_API_BASE_URL = `${AppSettings.getApiEndpoint()}/executions`;
+export const WORKFLOW_VERSIONS_API_BASE_URL = `${AppSettings.getApiEndpoint()}/version`;
 
 @Injectable({
   providedIn: "root",
 })
 export class WorkflowExecutionsService {
-  constructor(private http: HttpClient) {}
+  constructor(
+    private http: HttpClient,
+    private workflowActionService: WorkflowActionService,
+    private undoRedoService: UndoRedoService,
```

Comment:
in the preview mode, is the user able to move the operators around? if yes, then you need to disable persisting, and also disable the undo redo

Reply:
In preview mode, users are not allowed to move the operators.